src/jit/assertionprop.cpp

```diff
@@ -2215,9 +2215,14 @@ GenTreePtr Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTreePtr stmt,
 #ifdef _TARGET_64BIT_           
             if (vnStore->IsVNHandle(vnCns))
             {
-                newTree = gtNewIconHandleNode(value, vnStore->GetHandleFlags(vnCns));
-                newTree->gtVNPair = ValueNumPair(vnLib, vnCns);
-                newTree = optPrepareTreeForReplacement(tree, newTree);
+				// Don't perform constant folding that involves a handle that needs
+				// to be recorded as a relocation with the VM.
+				if (!opts.compReloc)
```

Comment:
With your suggestion I would have to call IsVNHandle twice: once inside IsRelocatableVNHandle and once outside. I don't think this refactoring is worth it.